packages/runfiles/runfiles.ts

```diff
@@ -23,20 +23,26 @@ export class Runfiles {
    */
   repoMappings: RepoMappings|undefined;
 
-  constructor(private _env = process.env) {
+  private _initialized = false;
+
+  constructor(private _env = process.env) {}
+
+  private _initialize() {
+    if (this._initialized) return;
+    this._initialized = true;
```

Comment:
This means the public methods such as `loadRunfilesManifest()` will only throw the first time, then `this._initialized = true` will prevent it from throwing again and the behaviour of `loadRunfilesManifest()` is then inconsistent.

Maybe we should keep the initialization in the `constructor()`, but move the `throw` into the public methods?

Reply:
I could also move the flag change to be at the end of the _initialize method so that it doesn't ever mark it as initialized if it throws in the middle of the initialization

Reply:
Ended up going with your initial suggestion and making sure that proper environment variables are set when the user calls `resolve()` either directly or as a result of calling another public method that calls it internally.